conductor_api/src/conductor/admin.rs

```diff
@@ -209,12 +209,11 @@ impl ConductorAdmin for Conductor {
                     .into(),
             },
         };
-        new_config.instances.push(new_instance_config);
-        new_config.check_consistency(&mut self.dna_loader)?;
-        let instance = self.instantiate_from_config(id, Some(&mut new_config))?;
+        self.config.instances.push(new_instance_config);
+        self.config.check_consistency(&mut self.dna_loader)?;
+        let instance = self.instantiate_from_config(id)?;
         self.instances
             .insert(id.clone(), Arc::new(RwLock::new(instance)));
-        self.config = new_config;
         self.save_config()?;
```

Comment:
Hm, this change seems problematic to me. The old code would jump out in line 213 if the new config is bogus and not change `self` at all. Now the new instance will always get added to `self.config.instances`.

I don't understand why we need to change these lines at all. What is wrong with the old code?